src/Libraries/CoreNodeModelsWpf/Controls/CustomSelectionControl.xaml

```diff
@@ -0,0 +1,170 @@
+                                    <Setter TargetName="Border" Property="Background" Value="Transparent"/>
+                                </Trigger>
+                            </ControlTemplate.Triggers>
+                        </ControlTemplate>
+                    </Setter.Value>
+                </Setter>
+            </Style>
+
+
+            <Style x:Key="RemoveButtonStyle" TargetType="{x:Type Button}">
+                <Setter Property="Foreground" Value="{StaticResource RemoveButtonColor}" />
+                <Setter Property="BorderThickness" Value="0"/>
+                <Setter Property="BorderBrush" Value="{x:Null}"/>
+                <Setter Property="Background" Value="{x:Null}"/>
+                <Setter Property="ClickMode" Value="Press" />
+                <Setter Property="Margin" Value="5,8,0,5"/>
+                <Setter Property="Height" Value="10" />
+                <Setter Property="Width" Value="10"/>
+                <Setter Property="VerticalAlignment" Value="Center"/>
+                <Setter Property="Template">
+                    <Setter.Value>
+                        <ControlTemplate TargetType="{x:Type Button}">
+                            <Border x:Name="border" BorderBrush="{TemplateBinding BorderBrush}" BorderThickness="{TemplateBinding BorderThickness}" Background="{TemplateBinding Background}" SnapsToDevicePixels="true">
+                                <ContentPresenter x:Name="contentPresenter" Focusable="False" HorizontalAlignment="{TemplateBinding HorizontalContentAlignment}" Margin="{TemplateBinding Padding}" RecognizesAccessKey="True" SnapsToDevicePixels="{TemplateBinding SnapsToDevicePixels}" VerticalAlignment="{TemplateBinding VerticalContentAlignment}"/>
+                            </Border>
+                        </ControlTemplate>
+                    </Setter.Value>
+                </Setter>
+                <Style.Triggers>
+                    <DataTrigger Binding="{Binding RelativeSource={RelativeSource Self}, Path=IsMouseOver}" Value="True">
+                        <Setter Property="Foreground" Value="{StaticResource HighlightedRemoveButtonColor}"/>
+                    </DataTrigger>
+                </Style.Triggers>
+            </Style>
+
+            <Style x:Key="RemoveButtonImage" TargetType="{x:Type Image}">
+                <Setter Property="Source">
+                    <Setter.Value>
+                        <DrawingImage>
+                            <DrawingImage.Drawing>
+                                <GeometryDrawing Brush="{Binding ElementName=RemoveBtn, Path=Foreground}">
+                                    <GeometryDrawing.Pen>
+                                        <Pen Brush="{Binding ElementName=RemoveBtn, Path=Foreground}" Thickness="5"/>
+                                    </GeometryDrawing.Pen>
+                                    <GeometryDrawing.Geometry>
+                                        <PathGeometry>
```

Comment:
does this align with other "remove"/"cancel" buttons in Dynamo - I would think this should look like the `remove` button on variable input nodes.?
FYI @Jingyi-Wen 
- I would also be a bit curious about any performance impact of drawing all these paths, if for example a graph contains dozens of these nodes with many entries each - maybe we can just replace this with an image which will only need to be loaded once - we have lots of "x" images floating around.

Reply:
⚠️ I also believe these styles should be moved the shared resource dictionary, embedding styles within the resource entry of a particular view means that for every view you're going to be instantiating this style again and again, increasing memory use for each node view, if instead you put this in the shared dictionary, you avoid that cost - please move it.

you are already importing one of the shared dictionaries above.

see:
https://docs.microsoft.com/en-us/windows/uwp/debug-test-perf/optimize-xaml-loading#resourcedictionary-in-a-usercontrol

Reply:
the relevant quote is:
```
A ResourceDictionary defined inside of a [UserControl](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.usercontrol) carries a penalty. The platform creates a copy of such a ResourceDictionary for every instance of the UserControl. If you have a UserControl that is used a lot, then move the ResourceDictionary out of the UserControl and put it at the page level.
```

Reply:
@mjkkirschner Also going to cover in follow up task